src/toxcore/dht/client.rs

```diff
@@ -135,25 +135,51 @@ impl Client {
                 IfAddr::V4(ref addr) => addr.broadcast,
                 _ => None,
         })
-        .map(|ipv4|
-            IpAddr::V4(ipv4)
-        ).collect()
+        .map(|addr|
+            IpAddr::V4(addr)
+        )
+        .collect()
+    }
+    // get broadcast addresses for host's network interfaces
+    fn get_ipv6_broadcast_addrs() -> Vec<IpAddr> {
+        let ifs = get_if_addrs::get_if_addrs().expect("no network interface");
+        ifs.iter().filter_map(|interface|
+            match interface.addr {
+                IfAddr::V6(ref addr) => addr.broadcast,
+                _ => None,
+        })
+        .map(|addr|
+            IpAddr::V6(addr)
+        )
+        .collect()
     }
     // get broadcast addresses for Ipv4 and Ipv6
     fn get_broadcast_addresses() -> Vec<IpAddr> {
-        let mut res = Vec::new();
-        let ipv4s = Client::get_ipv4_broadcast_addrs();
+        let mut res = Vec::<IpAddr>::new();
 
-        res.extend(ipv4s.iter());
+        // there will be a cases which are host's interface may have IPv4 only,
+        // IPv6 only, both IPv4 and IPv4
+        // If ipv4 only, broadcast to only Ipv4
+        // If ipv6 only, braodcast to only Ipv6
+        // If both, braodcast to both addresses
+        let ips = Client::get_ipv4_broadcast_addrs();
+        res.extend(ips.iter());
+
+        let ips = Client::get_ipv6_broadcast_addrs();
+        res.extend(ips.iter());
+
+        if !res.iter().filter(|ip| ip.is_ipv4()).next().is_none() {
```

Comment:
It seems to me it's not the right way to check type of broadcast address. It should depend on socket type.

Reply:
Yes. you are right.
Toxcore runs as Ipv4 or ipv6 mode, do not run as both mode at the same time.